typewriter/fixes/fixer_utils.py

```diff
@@ -133,3 +140,254 @@ def touch_import(package, name, node):
+
+    def __hash__(self):
+        return hash((self.node.get_lineno(),
+                     self.node.depth(),
+                     self.node.__str__()))
+
+
+@cache
+def _get_import_info_cachable(hashable_node):
+    # type: (HashableNode) -> ImportInfo
+    """
+    We cache using hashable_node as a key and analyze hashable_node.node
+
+    If node is a valid import_stmt, this will return a named tuple for each component of the
+    statement
+        ex: from a.b import c as d, e  => (imports=[(import_name='c',
+                                                     binding_name='d'),
+                                                    (import_name='e',
+                                                     binding_name='e')],
+                                           package='a.b',
+                                           node=node)
+
+    Since we will regularly iterate over the list import node for their information when
+    searching for binding or import matches, we cache the results of this function for
+    quick and non-redundant lookups
+
+    Parameters
+    -----------
+    hashable_node : HashableNode
+
+    Return
+    -----------
+    ImportInfo
+    """
+    # Get the node from the hashable_node wrapper
+    node = hashable_node.node
+
+    def handle_name(node, allow_dotted=True):
+        if (node.type == syms.dotted_as_name and allow_dotted)\
+                or node.type == syms.import_as_name:
+            # [Grammar] dotted_as_name: dotted_name ['as' NAME]
+            package = import_name = binding_name = None
+            name = node.children[0]
+            binding_name = str(node.children[2]).strip()
+            if name.type == syms.dotted_name and allow_dotted:
+                # [Grammar] dotted_name: NAME ('.' NAME)*
+                # We don't need the third field since we know the alias will be the binding_name
+                package, import_name, _ = decompose_name(name)
+            elif name.type == token.NAME:
+                import_name = name.value
+            return package, import_name, binding_name
+        if allow_dotted:
+            # As long as we aren't ignoring dotted names
+            # If there was no "as", we know this is dotted_name
+            # [Grammar] dotted_name: NAME ('.' NAME)*
+            return decompose_name(node)
+        # If we ignored dotted names, then we are left with only a token.NAME node, cast and return
+        name = str(node).strip()
+        return None, name, name
+
+    package = None
+    imports = set()  # type: Set[ImportPairing]
+    if node.type == syms.import_name:
+        # [Grammar]: import_name: 'import' dotted_as_names
+        as_name = node.children[1]
+        if as_name.type in (syms.dotted_as_names, syms.import_as_names):
+            # [Grammar]: dotted_as_names: dotted_as_name (',' dotted_as_name)*
+            for child in as_name.children:
+                package, import_name, binding_name = handle_name(child)
```

Comment:
I think we either need to filter out `','` results in this loop, or inside of `handle_name`